### PR TITLE
Adds built-in Franceschini (2018) and Saldana-Lopez (2021) EBL models

### PR DESCRIPTION
**Pull request by cosimoNigro**
Paired with [PR 29 in gammapy-data](https://github.com/gammapy/gammapy-data/pull/29) this PR adds the EBL models of Franceschini (2018) and Saldana-Lopez (2021).

Solves #4055.

What to do with the tests? It seems to me that no test uses all the built-in models available: there is one testing that a `TemplateSpectralModel` [can read one of the EBL file](https://github.com/gammapy/gammapy/blob/main/gammapy/modeling/models/tests/test_spectral.py#L540) and [one that checks its extrapolation](https://github.com/gammapy/gammapy/blob/main/gammapy/modeling/models/tests/test_spectral.py#L589). Then another [one tests the effect of the absorption](https://github.com/gammapy/gammapy/blob/main/gammapy/modeling/models/tests/test_spectral.py#L550). There are some files untouched by the tests, should we add a test involving all the built-in models?

### CODE REVIEW

**Comment by QRemy** [comment]
Thanks @cosimoNigro.
In the read_builtin method,  should we rename the previous ones to also include the year in the name ?
(That's backward incompatible but may be nicer at the end).

**Comment by registerrier** [comment]
@cosimoNigro could you rebase and use the registry introduced in #4214 ?

**Review by QRemy — APPROVED** [review]
Thanks @cosimoNigro, I made the necessary change after the rebase, so this can be merged if the models files are in $GAMMAPY_DATA

**Pull request merged**
